code/game/gamemodes/gang/gang.dm

```diff
@@ -0,0 +1,366 @@
+				cops_to_send = /datum/antagonist/ert/families/beatcop/swat
+			if(FOUR_STARS_LOW to FIVE_STARS_LOW-1)
+				team_size = 8
+				cops_to_send = /datum/antagonist/ert/families/beatcop/fbi
+			if(FIVE_STARS_LOW to INFINITY)
+				team_size = 10
+				cops_to_send = /datum/antagonist/ert/families/beatcop/military
+
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Do you want to help clean up crime on this station?", "deathsquad", null)
+
+	if(candidates.len > 0)
+		//Pick the (un)lucky players
+		var/numagents = min(team_size,candidates.len)
+
+		var/list/spawnpoints = GLOB.emergencyresponseteamspawn
+		var/index = 0
+		while(numagents && candidates.len)
+			var/spawnloc = spawnpoints[index+1]
+			//loop through spawnpoints one at a time
+			index = (index + 1) % spawnpoints.len
+			var/mob/dead/observer/chosen_candidate = pick(candidates)
+			candidates -= chosen_candidate
+			if(!chosen_candidate.key)
+				continue
+
+			//Spawn the body
+			var/mob/living/carbon/human/cop = new(spawnloc)
+			chosen_candidate.client.prefs.copy_to(cop)
+			cop.key = chosen_candidate.key
+
+			//Give antag datum
+			var/datum/antagonist/ert/ert_antag = new cops_to_send
+
+			cop.mind.add_antag_datum(ert_antag)
+			cop.mind.assigned_role = ert_antag.name
+			SSjob.SendToLateJoin(cop)
+
+			//Logging and cleanup
+			log_game("[key_name(cop)] has been selected as an [ert_antag.name]")
+			numagents--
+		cops_arrived = TRUE
+		addtimer(CALLBACK(src, .proc/end_hostile_sit), 10 MINUTES)
+		return TRUE
+	else
+		cops_arrived = TRUE
+		addtimer(CALLBACK(src, .proc/end_hostile_sit), 10 MINUTES)
+		return FALSE
+
+/datum/game_mode/gang/proc/end_hostile_sit()
+	SSshuttle.clearHostileEnvironment(src)
+
+/datum/game_mode/gang/proc/check_tagged_turfs()
+	for(var/obj/effect/decal/cleanable/crayon/gang/tag in GLOB.gang_tags)
+		if(tag.my_gang)
+			tag.my_gang.adjust_points(50)
+		CHECK_TICK
+
+/datum/game_mode/gang/proc/check_gang_clothes() // TODO: make this grab the sprite itself, average out what the primary color would be, then compare how close it is to the gang color so I don't have to manually fill shit out for 5 years for every gang type
+	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
+		if(!H.mind || !H.client)
+			continue
+		var/datum/antagonist/gang/is_gangster = H.mind.has_antag_datum(/datum/antagonist/gang)
+		for(var/clothing in list(H.head, H.wear_mask, H.wear_suit, H.w_uniform, H.back, H.gloves, H.shoes, H.belt, H.s_store, H.glasses, H.ears, H.wear_id))
+			if(is_gangster)
+				if(is_type_in_list(clothing, is_gangster.acceptable_clothes))
+					is_gangster.add_gang_points(10)
+			else
+				for(var/datum/team/gang/gang_clothes in gangs)
+					if(is_type_in_list(clothing, gang_clothes.acceptable_clothes))
+						gang_clothes.adjust_points(5)
+
+		CHECK_TICK
+
+/datum/game_mode/gang/proc/check_rollin_with_crews()
+	for(var/area/A in GLOB.sortedAreas)
+		var/list/gang_members = list()
+		for(var/mob/living/carbon/human/H in A)
```

Comment:
I think approaching this the other way around and looping through all players makes more sense then looping through the areas.

Reply:
it'd be faster to just loop through gangbangers, get their areas, then loop that list of areas I think
that way im not looping the area list or the player list